sample/src/main/java/org/aerogear/graphqlandroid/data/UserData.kt

```diff
@@ -0,0 +1,152 @@
+package org.aerogear.graphqlandroid.data
+
+import android.content.Context
+import android.util.Log
+import com.apollographql.apollo.ApolloCall
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.cache.http.HttpCachePolicy
+import com.apollographql.apollo.cache.normalized.ApolloStore
+import com.apollographql.apollo.exception.ApolloException
+import org.aerogear.graphqlandroid.*
+import org.aerogear.graphqlandroid.model.Task
+
+class UserData {
+
+
+    val TAG = javaClass.simpleName
+
+    lateinit var apolloStore: ApolloStore
+
+    fun getTasks(context: Context): ArrayList<Task> {
+
+        var noteslist = arrayListOf<Task>()
+
+        Log.e(TAG, "inside getTasks")
+
+        val client = Utils.getApolloClient(context)?.query(
+            AllTasksQuery.builder().build()
+        )?.httpCachePolicy(HttpCachePolicy.CACHE_FIRST)
+
+        Log.e(TAG, " getActiveCallsCount : ${Utils.getApolloClient(context)?.activeCallsCount()}")
+
+        apolloStore = Utils.getApolloClient(context)?.apolloStore()!!
+
+        Log.e(TAG, " apolloStore : ${Utils.getApolloClient(context)?.apolloStore()}")
+
+        client?.enqueue(object : ApolloCall.Callback<AllTasksQuery.Data>() {
```

Comment:
This is where problem lands. We do single query with cache first policy. 
THis makes no sense as.. 

1) we get result from cache
2) we never wait for another result from network
3) we are not going to watch other cache changes

See example: 
https://github.com/apollographql/apollo-android/blob/1829a241f99c425abe0fe445282560f0749045ed/apollo-integration/src/test/java/com/apollographql/apollo/internal/QueryRefetchTest.java#L150-L162

Sadly this part is not fully documented in repo.

Reply:
@wtrocki It's written that in case of Cache first, If the response doesn't exist or is expired, then fetch a response from the network. Here expired means outdated?

Reply:
There are no expiration policies implemented yet so it is only when cache is empty. More about workaround for expiration policies: https://github.com/apollographql/apollo-feature-requests/issues/4#issuecomment-498740069